packages/editor/src/components/block-list/block.js

```diff
@@ -86,6 +88,13 @@ export class BlockListBlock extends Component {
 	}
 
 	componentDidUpdate( prevProps ) {
+		if ( this.state.isHandlingAltF10 && ! this.props.isTypingWithinBlock ) {
+			// When the component updates and the user is not typing within the block
+			// we know that handling alt+f10 keypress is finished (the necessary child blocks are rendered).
+			this.setState( {
```

Comment:
Would it be easy enough to use an instance variable `this.isForcingContextualToolbar`? This would avoid the need for a second re-render.

Reply:
We can use an instance variable the catch is that we need to force a re-render when changing the variable to true. I updated to use this approach.